### PR TITLE
Added Gfx and Gfx2d setup

### PR DESCRIPTION
**Pull request by bvssvni**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -28,13 +37,42 @@ pub struct PistonWindow<T = ()> {
     pub app: Rc<RefCell<T>>,
 }
 
+/// Contains Gfx data.
+pub struct Gfx {
+    /// The device.
+    pub device: gfx_device_gl::Device,
```
Comment (kvark) [inline src/lib.rs]: Looks like you could have used `gfx::Canvas` here.

**Comment by kvark** [comment]
Ok, so piston-window is gfx-only? Or am I missing something? Leaving glium out of scope would be a big mistake, to my mind.

**Comment by bvssvni** [comment]
@kvark Why is that a big mistake?

**Comment by kvark** [comment]
I was misunderstanding the scope of piston-window. All good now!

**Pull request merged**